dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/filter/AccessLogFilter.java

```diff
@@ -141,6 +141,7 @@ private void writeLogQueueToFile(String accessLog, Queue<AccessLogData> logQueue
                 processWithAccessKeyLogger(logQueue, file);
             }
         } catch (Exception e) {
+            logQueue.poll();
```

Comment:
It will be better to modify the traversal method of the queue.

Reply:
> It will be better to modify the traversal method of the queue.

Yes, I think so too, but I'm afraid that changing too much code might not be acceptable

Reply:
> It will be better to modify the traversal method of the queue.

And if you modify the queue traversal methods so processWithAccessKeyLogger two overloaded methods need to be modified

Reply:
I'm sorry, maybe I didn't express it clearly, you should handle exceptions in processWithAccessKeyLogger, and you also need to dequeue operations. If the queue is not empty, the `while` operation will cause the thread to be blocked all the time.
@106umao